gapic/generator/generator.py

```diff
@@ -274,6 +274,10 @@ def _render_template(
                 if (
                         skip_subpackages
                         and service.meta.address.subpackage != api_schema.subpackage_view
+                        or
+                        'grpc' in template_name and 'grpc' not in opts.transport
```

Comment:
This little logic chunk could be lifted up and generalized, maybe something like
```python
def is_undesired_transport(template_name: str, transport: str, opts) - > bool:
    return transport in template_name and transport not in opts.transport
```

Also: I'm still leaning towards "http" instead of "rest" for the template naming scheme. It just seems more descriptive of what the protocol is.